source/mbed_trace.c

```diff
@@ -454,8 +454,12 @@ void mbed_vtracef(uint8_t dlevel, const char* grp, const char *fmt, va_list ap)
 
 end:
     if ( m_trace.mutex_release_f ) {
-        while (m_trace.mutex_lock_count > 0) {
-            m_trace.mutex_lock_count--;
+        // Store the mutex lock count to temp variable so that it won't get
```

Comment:
There's these where the mutex is acquired: https://github.com/ARMmbed/mbed-trace/blob/6cf459bca0fc744ef52be428100e896cb861f2b6/source/mbed_trace.c#L480

But agreed, more wouldn't hurt.